design/20190708.certificate-request-crd.md

```diff
@@ -174,6 +188,190 @@ type ObjectReference struct {
+approves this request, the violations which caused the request to be denied, or
+the team to who manually approved the request.
+
+When a CertificateRequest has been Denied, it is the responsibility of the
+referenced issuer to then add a Ready condition with the status of "False",
+along with a relevant Reason and Message.
+
+
+##### RBAC
+
+Approved and Denied conditions are set by requesting against the `/status`
+endpoint of the CertificateRequest resource. This is a divergence of the
+certificates.k8s.io CertificateSigningRequest resource that utilises `approval`,
+as CustomResourceDefinitions may [only define `/status` or `/scale`
+sub-resources](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#subresources).
+Approvers must therefore have permissions to update the status sub-resource of
+CertificateRequests. Approvers missing this permissions will have their request
+rejected by the API server.
+
+Setting the Approved or Denied conditions are restricted by the approver having
+sufficient RBAC permissions. These permissions are based upon the request
+itself- specifically the request's IssuerRef:
+
+```yaml
+apiGroups: ["cert-manager.io"]
+resources: ["signers"]
+verbs: ["approve"]
+resourceNames:
+ # namesapced signers
+ - "<signer-resource-name>.<signer-group>/<signer-namespace>.<signer-name>"
+ # cluster scoped signers
+ - "<signer-resource-name>.<signer-group>/<signer-name>"
+ # all signers of this resource name
+ - "<signer-resource-name>.<signer-group>/*"
+```
+
+An example ClusterRole that would grant the permissions to set the Approve and
+Denied conditions of CertificateRequests that reference the cluster scoped
+`myissuers` external issuer, in the group `my-example.io`, with the name `myapp`:
+
+```
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: my-example-io-my-issuer-myapp-approver
+rules:
+  - apiGroups: ["cert-manager.io"]
+    resources: ["signers"]
+    verbs: ["approve"]
+    resourceNames: ["myissuers.my-example.io/myapp"]
+```
+
+These permissions are enforced using a
+[SubjectAccessReview](https://github.com/kubernetes/kubernetes/blob/b11d0fbdd58394a62622787b38e98a620df82750/pkg/apis/authorization/types.go#L27)
+that is performed in the cert-manager Webhook component when a user attempts to
+set the Approve or Denied condition. The user of the approver is tested along
+with the IssuerRef of the CertificateRequest.
+
+If the approver does not have sufficient permissions defined above to set the
+Approved or Denied conditions, the request will be rejected by the cert-manager
+validating admission webhook.
+
+###### Scope
+The RBAC permissions *must* be granted at the cluster scope (i.e.
+ClusterRoleBinding). This matches the same scoping behaviour as
+[certificates.k8s.io
+CertificateSigningRequests](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#authorization).
```

Comment:
Why should it only be at the cluster scope?

Reply:
This is the same behaviour as the upstream CSR resource, which is cluster scoped. The namespace is encoded within the signer ref (exmaple.io/my-namespace.bar)